### PR TITLE
feat: add support for comparing CRAN versions

### PR DESCRIPTION
**Pull request by G-Rath**
Part of #642

See https://github.com/G-Rath/osv-detector/pull/235 for the journey I went on with R for this

### CODE REVIEW

File: scripts/generators/generate-cran-versions.R [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+      package <- affected$package$name
+
+      if (!(package %in% names(result))) {
+        result[[package]] <- list()
+      }
+
+      for (version in affected$versions) {
+        tryCatch(
+          {
+            as.package_version(version)
```
Comment (cuixq) [inline scripts/generators/generate-cran-versions.R:36]: what does `as` here mean?

Comment (G-Rath) [reply scripts/generators/generate-cran-versions.R:36]: From what I understand (which is very little in this space 😅) it's R's version of casting: https://www.rdocumentation.org/packages/methods/versions/3.0.3/topics/as

So R has this [`package_version`](https://astrostatistics.psu.edu/su07/R/html/base/html/package_version.html) class, and we can make one of them by doing `as.package_version(...)` which... apparently we can also do by just calling `package_version(...)`?

It sounds like among other things [`as` is more performant as it does a quick return if the arg is already the right type](https://stackoverflow.com/questions/39279238/why-use-as-factor-instead-of-just-factor)

File: internal/semantic/version-cran.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
+package semantic
```
Comment (oliverchang) [inline internal/semantic/version-cran.go:1]: Is there an upstream documentation page for the versioning that we can link from here?

**Review by oliverchang — COMMENTED** [review]
very nice!!

**Review by oliverchang — APPROVED** [review]

**Comment by G-Rath** [comment]
also just a note: from what I can tell this is actually more "R" rather than "CRAN" aka this is exactly the same for the Bioconductor ecosystem, but I've not worried about that at all because 1. `semantic` is internal and 2. there are advisories for any Bioconductor package.

I think it should be pretty easy if just slightly "special-case-y" to handle if/when there's Bioconductor advisories (this will also apply to #668)

**Pull request merged**